### ISSUE TITLE
MetadataAsSource files not added to MetadataAsSource workspace (and workspace is not registered

### ISSUE THREAD

**Issue by dibarbet** [dotnet/vscode-csharp#5787]
Currently it looks like all metadata as source files are going into the LSP misc workspace. This appears to be because TryAddDocumentToWorkspace never gets called. In VS this is called when the file is opened (the misc files workspace sees it and puts the document in the workspace). Since a solution is never created in the metadata workspace because the file is never added, the metadata workspace is also not registered.

**Comment by dibarbet** [comment]
Scenarios here
1.  Go to definition from metadata references
2.  Document outline - https://github.com/dotnet/vscode-csharp/issues/6341

**Comment by daniel-tonic** [comment]
Is there any update on this?  This seems like a pretty big miss from a feature-parity perspective and it's been several months since this ticket was updated.

**Comment by hsm207** [comment]
Just encountered #6021 but only for some projects in a solution.

I'm on version 2.22.5 and everytime I reload the window, the C# output pane shows this:

```
Using dotnet configured on PATH
Dotnet path: /usr/share/dotnet/dotnet
Activating C# + C# Dev Kit...
waiting for named pipe information from server...
[stdout] {"pipeName":"/tmp/c8da874b.sock"}
received named pipe information from server
attempting to connect client to server...
client has connected to server
[Info  - 3:42:10 AM] [Program] Language server initialized
[Error - 3:42:11 AM] Request textDocument/semanticTokens/range failed.
  Message: Unsupported extension '' and LSP language id ''
  Code: -32000 
[object Object]
[Error - 3:42:11 AM] Request textDocument/semanticTokens/range failed.
  Message: Unsupported extension '' and LSP language id ''
  Code: -32000 
[object Object]
```

**Comment by whitecostume** [comment]
Still not solved?

**Comment by dibarbet** [comment]
This should be available in the next prerelease update to the extension.  Will update here once it is published.

**Comment by dibarbet** [comment]
This should now be available in v2.41.26 (pre-release).

**Comment by dibarbet** [comment]
@SherbetHead yeah there is an issue with sourcelink in the latest version.  We're working on a fix and should have it out this week.

**Comment by HunterZ** [comment]
What's the status on this? It's frustrating only being able to navigate one level deep into decompiled code.

**Comment by dibarbet** [comment]
@HunterZ this should be available in the latest versions of the extension (either release or prerelease).  If you're having problems on the latest version, would you mind filing a new issue?  Would be helpful to include the C# logs - https://github.com/dotnet/vscode-csharp/blob/main/SUPPORT.md#collecting-general-logs as well as any information on where the symbols you're navigating to are coming from.

**Comment by HunterZ** [comment]
After having never worked, this suddenly seems to have started working a few days ago. I guess it just took 2-3 months to get into a release version.
